qiskit_optimization/translators/ising.py

```diff
@@ -166,90 +153,69 @@ def from_ising(
         )
 
     quad_prog = QuadraticProgram()
-
-    # add binary variables
-    for i in range(qubit_op.num_qubits):
-        quad_prog.binary_var(name="x_{0}".format(i))
-
-    # Create a QUBO matrix
-    # The Qubo matrix is an upper triangular matrix.
-    # Diagonal elements in the QUBO matrix are for linear terms of the qubit operator.
-    # The other elements in the QUBO matrix are for quadratic terms of the qubit operator.
-    qubo_matrix = zeros((qubit_op.num_qubits, qubit_op.num_qubits))
+    quad_prog.binary_var_list(qubit_op.num_qubits)
 
     if not isinstance(qubit_op, SummedOp):
         pauli_list = [qubit_op.to_pauli_op()]
     else:
         pauli_list = qubit_op.to_pauli_op()
 
+    # prepare a matrix of coefficients of Pauli terms
+    # `pauli_coeffs_diag` is the diagonal part
+    # `pauli_coeffs_triu` is the upper triangular part
+    pauli_coeffs_diag = [0.0] * qubit_op.num_qubits
+    pauli_coeffs_triu = {}
+
     for pauli_op in pauli_list:
         pauli_op = pauli_op.to_pauli_op()
         pauli = pauli_op.primitive
         coeff = pauli_op.coeff
-        # Count the number of Pauli Zs in a Pauli term
-        lst_z = pauli.z.tolist()
-        z_index = [i for i, z in enumerate(lst_z) if z is True]
+
+        if coeff.imag != 0.0:
+            raise QiskitOptimizationError(f"Imaginary coefficient exists: {pauli_op}")
```

Comment:
I added a error check with imaginary coefficients.

Reply:
I have seen situations where sometimes we have aspects that are e-17 or very small. If it has a tiny imaginary part due some some computation rounding should we just ignore it. e.g `if abs(coeff.imag) > 1e-15` or whatever suitable threshold only then raise an error. What do you think?

Reply:
Good point. I use `math.isclose` for zero check. fc9afe56d97a5a0ef3428c